delphi-checks/src/main/java/au/com/integradev/delphi/checks/RedundantInheritedCheck.java

```diff
@@ -0,0 +1,144 @@
+import org.sonar.plugins.communitydelphi.api.ast.RoutineImplementationNode;
+import org.sonar.plugins.communitydelphi.api.ast.StatementListNode;
+import org.sonar.plugins.communitydelphi.api.ast.StatementNode;
+import org.sonar.plugins.communitydelphi.api.ast.utils.ExpressionNodeUtils;
+import org.sonar.plugins.communitydelphi.api.check.DelphiCheck;
+import org.sonar.plugins.communitydelphi.api.check.DelphiCheckContext;
+import org.sonar.plugins.communitydelphi.api.check.FilePosition;
+import org.sonar.plugins.communitydelphi.api.reporting.QuickFix;
+import org.sonar.plugins.communitydelphi.api.reporting.QuickFixEdit;
+import org.sonar.plugins.communitydelphi.api.symbol.declaration.RoutineNameDeclaration;
+import org.sonar.plugins.communitydelphi.api.token.DelphiTokenType;
+
+@Rule(key = "RedundantInherited")
+public class RedundantInheritedCheck extends DelphiCheck {
+  private static final String MESSAGE = "Remove this redundant inherited call.";
+
+  @Override
+  public DelphiCheckContext visit(RoutineImplementationNode routine, DelphiCheckContext context) {
+    for (DelphiNode violationNode : findViolations(routine)) {
+      context
+          .newIssue()
+          .onNode(violationNode)
+          .withMessage(MESSAGE)
+          .withQuickFixes(getQuickFixes(violationNode))
+          .report();
+    }
+    return super.visit(routine, context);
+  }
+
+  private static List<QuickFix> getQuickFixes(DelphiNode violationNode) {
+    DelphiNode statement = violationNode;
+    DelphiNode parent = violationNode.getParent();
+    while (!(parent instanceof StatementListNode)) {
+      statement = parent;
+      parent = parent.getParent();
+      if (parent == null) {
+        return null;
+      }
+    }
+    StatementListNode statementListNode = (StatementListNode) parent;
+    int statementIndex = statementListNode.getStatements().indexOf(statement);
+
+    int startLine = statement.getBeginLine();
+    int startCol = statement.getBeginColumn();
+    int endLine = statement.getEndLine();
+    int endCol = statement.getEndColumn();
+
+    if (statementListNode.getStatements().size() > statementIndex + 1) {
+      StatementNode statementNode = statementListNode.getStatements().get(statementIndex + 1);
+      endLine = statementNode.getBeginLine();
+      endCol = statementNode.getBeginColumn();
+    } else {
+      if (statementIndex > 0) {
+        StatementNode previousStatement = statementListNode.getStatements().get(statementIndex - 1);
+        FilePosition startPosition = skipTrailingSemicolon(statementListNode, previousStatement);
+        startLine = startPosition.getBeginLine();
+        startCol = startPosition.getBeginColumn();
+      }
+
+      FilePosition endPosition = skipTrailingSemicolon(statementListNode, statement);
+      endLine = endPosition.getEndLine();
+      endCol = endPosition.getEndColumn();
+    }
+
+    return List.of(
+        QuickFix.newFix("Remove the redundant inherited call")
```

Comment:
```suggestion
        QuickFix.newFix("Remove redundant inherited call")
```

Quick fixes have limited screen real estate, so ideally they're as concise as possible.